service/history/handler.go

```diff
@@ -1854,6 +1854,126 @@ func (h *Handler) UpdateWorkflowExecution(
 	return engine.UpdateWorkflowExecution(ctx, request)
 }
 
+func (h *Handler) ReadHistoryBranch(
+	ctx context.Context,
+	request *historyservice.ReadHistoryBranchRequest,
+) (_ *historyservice.ReadHistoryBranchResponse, retError error) {
+	defer log.CapturePanic(h.logger, &retError)
+	h.startWG.Wait()
```

Comment:
spoke offline... this should be dealt with separately

Reply:
To add some more context: we decided this change should be done wholesale (on each method here) since it would be weird to have it in just this one. Additionally, we concluded that it shouldn't ride on this PR because it could have unintended side-effects, even if it doesn't appear that way, so it would warrant testing which would add too much to this effort.